tree/binary-search-tree/binary-search-tree-node.js

```diff
@@ -0,0 +1,111 @@
+const { BinaryTreeNode } = require('../binary-tree-node')
+
+class BinarySearchTreeNode extends BinaryTreeNode {
+  constructor(value) {
+    super(value)
+  }
+
+  insert(value) {
+    // tree is empty
+    if (this.value === null) {
+      this.value = value
+      return this
+    }
+
+    if (value < this.value) {
+      if (this.left) {
+        return this.left.insert(value)
+      }
+
+      const newNode = new BinarySearchTreeNode(value)
+      this.setLeft(newNode)
+
+      return newNode
+    }
+
+    if (value > this.value) {
+      if (this.right) {
+        return this.right.insert(value)
+      }
+
+      const newNode = new BinarySearchTreeNode(value)
+      this.setRight(newNode)
+
+      return newNode
+    }
+
+  }
+
+  delete(value) {
+    const nodeToRemove = this.find(value)
+    if (!nodeToRemove) return false
+    let { parent } = nodeToRemove
+    
+    // nodeToRemove has no children
+    if (nodeToRemove.left === null && nodeToRemove.right === null) {
```

Comment:
It's better to extract this into a separate function to check whether a node is a leaf or not.